### PR TITLE
Task#60886 - Create a room

### PR DESCRIPTION
**Pull request by letrungkien**
## Ticket

https://dev.framgia.com/redmine/issues/60886
## Summary

Create room with basic information
- Room name
- Room description(some detail infomation such as location)
- Room capacity (number seat in room)

### CODE REVIEW

File: app/scripts/app.js [hunk 1]
```diff
@@ -57,13 +57,18 @@
         controller: 'ContactController',
         controllerAs: 'contact'
       })
+      .when('/rooms/new', {
+        templateUrl: 'views/rooms/new.html',
+        controller: 'RoomsController',
+        controllerAs: 'rooms'
```
Comment (yeuem1vannam) [inline app/scripts/app.js]: This representative name should be singularize

File: app/scripts/controllers/rooms_controller.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  function RoomsController($scope, $location, DataFactory){
+    $scope.room = {};
+    $scope.room.roomName = '';
+    $scope.room.roomDescription = '';
+    $scope.room.roomCapacity = 0;
+
+    $scope.createRoom = function(){
+      var roomsRef = new DataFactory('rooms');
+      var roomName = $scope.room.roomName;
```
Comment (xuanduc987) [inline app/scripts/controllers/rooms_controller.js]: Maybe we could move these logic to its own Service/Factory?

Comment (yeuem1vannam) [inline app/scripts/controllers/rooms_controller.js]: I agree

File: app/scripts/controllers/rooms_controller.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  'use strict';
+
+  angular
+    .module('app.controllers')
+    .controller('RoomsController', RoomsController);
+
+  RoomsController.$inject = ['$scope', '$location', 'DataFactory'];
+
+  function RoomsController($scope, $location, DataFactory){
+    $scope.room = {};
```
Comment (yeuem1vannam) [inline app/scripts/controllers/rooms_controller.js]: To take advantage of `controllerAs` syntax, I think you don't have to use `$scope` for data holding

File: app/views/rooms/new.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  <div class="form-group">
+    <label for="roomCapacity" class="col-sm-2 control-label">Room capacity:</label>
+    <div class="col-sm-4">
+      <input class="form-control" type="number" id="roomCapacity" ng-model="room.roomCapacity" placeholder="10"/>
+    </div>
+  </div>
+  <div class="form-group">
+    <div class="col-sm-offset-2 col-sm-10">
+      <button type="button" ng-click="createRoom()" class="btn btn-small btn-primary">Create</a>
```
Comment (yeuem1vannam) [inline app/views/rooms/new.html]: Should be changed follow `controllerAs` syntax

**Comment by yeuem1vannam** [comment]
TODO:
- [ ] Class / function description
- [x] `controllerAs` syntax

File: app/scripts/services/rooms_service.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    .module('app.services')
+    .service('RoomsService', RoomsService);
+
+  RoomsService.$inject = ['DataFactory'];
+
+  function RoomsService(DataFactory) {
+    this.create = create;
+
+    function create(room) {
+      var roomsRef = new DataFactory('rooms');
```
Comment (yeuem1vannam) [inline app/scripts/services/rooms_service.js]: Don't use `new` with `DataFactory`

**Comment by ghost** [comment]
Please rebase 6 commits to 1

**Comment by yeuem1vannam** [comment]
:o:K

**Pull request merged**